### ISSUE TITLE
FAR CLIPPING RANGE being controlled and overridden by world.

### ISSUE THREAD

**Issue by Zefnoly** [Yellow-Dog-Man/Resonite-Issues#2000]
### Is your feature request related to a problem? Please describe.

I am working on a world where the default far clipping range is too small. I need it only a little bit bigger.

Far clipping being controlled by individual avatars makes little sense and creates a big hurdle for certain world creations with 3D skyboxes and large background structures.

I feel like the avatarrendersettings only should control nearclip. Or have its far clipping overriden by a new component called (worldrendersettings) which would control far clipping of every avatar and streaming camera to ensure your world works as intended for everyone. Im not the only person with this issue and it puts a certain annoying limitation to creators as it is right now.

### Describe the solution you'd like

Maybe something like a new component like a "world render settings" that overrides far clipping range of avatars and streaming cams to ensure people joining your world sees it correctly?

### Describe alternatives you've considered

Completely replacing skyboxes and large background structures with subpar 2D skybox that is not as dynamic and much worse to experience in VR

### Additional Context

Basically, the default far clipping range is very limiting for world creators right now.

### Requesters

Zefnoly

**Comment by Zefnoly** [comment]
Potentially... Allowing a new render layer for skybox elements that are behind everything else in an easy way, would also solve a lot of this, and allow for small skybox elements, to still appear far away by simply being rendered before everything else every frame.

**Comment by shiftyscales** [comment]
Control over both near, and far clipping settings is important as they work in tandem. E.g. having both too small of a near clip, and too far of a far clip value can cause graphical issues as the floating point precision is spread over too wide a range.

AvatarRenderSettings were implemented at a time before other components like RenderTransformOverride existed, and were (and still are) useful as a means of culling out any geometry that gets too close to the user's view.

I'd suggest if you wish to make use of a world with a significantly increased far clipping range you also compensate by raising the near clipping value and have a sign/instruction posted somewhere in the world indicating that the experience was intended to be used with particular near/far clip settings.

Alternatively, there are several ways in which you can provide a pre-configured avatar that meets the needs of the world to users that join it depending on the needs of the experience.

> subpar 2D skybox

Stereoscopic textures are supported on the skybox component, so you can make a stereoscopic skybox if needed to bridge the gap.

> allow for small skybox elements, to still appear far away by simply being rendered before everything else every frame

You can mess with the render queue, and z-sorting properties on the material in order to get it to render behind other materials in the scene.

**Comment by Zefnoly** [comment]
> Control over both near, and far clipping settings is important as they work in tandem. E.g. having both too small of a near clip, and too far of a far clip value can cause graphical issues as the floating point precision is spread over too wide a range.
> 
> AvatarRenderSettings were implemented at a time before other components like RenderTransformOverride existed, and were (and still are) useful as a means of culling out any geometry that gets too close to the user's view.
> 
> I'd suggest if you wish to make use of a world with a significantly increased far clipping range you also compensate by raising the near clipping value and have a sign/instruction posted somewhere in the world indicating that the experience was intended to be used with particular near/far clip settings.
> 
Im not even talking about significant change. like 1.5x or 2x
> Alternatively, there are several ways in which you can provide a pre-configured avatar that meets the needs of the world to users that join it depending on the needs of the experience.
> 
This isnt a good solution in most cases, preferably you want your world joinable...
> > subpar 2D skybox
> 
> Stereoscopic textures are supported on the skybox component, so you can make a stereoscopic skybox if needed to bridge the gap.
> 
In usecases where you work with dynamic skyboxes that takes use of 3D rendered objects (planetary bodies for example), you can't simply use stereoscopic textures... Especially if you have moving sun, cloud layer, atmospheric effects etc... Or very difficult to do.
> > allow for small skybox elements, to still appear far away by simply being rendered before everything else every frame
> 
> You can mess with the render queue, and z-sorting properties on the material in order to get it to render behind other materials in the scene.

Not sure how to adjust z-buffer properties. And adjusting the render queue only seem to affect how transparent shaders are rendered with basic testing. I will try to look into it...

Another solution would be an easy way to have another render layer to isolate 3D objects intended for the skybox rendering that is more straight forward. Like other games do, for example Kerbal Space Program uses a different render layer for the skybox so you can have far away looking objects in the skybox rendered in 3D, without having to change the clipping planes.

**Comment by shiftyscales** [comment]
Seeking input from @Geenz.

**Comment by zahndy** [comment]
This or #3636 could fix the issues I have with my map which has a incredibly big skybox/surrounding (>20km radius)

![Image](https://github.com/user-attachments/assets/391c6824-3b8b-4d40-bca2-f36033794dbb)
